tests/manage/mcg/test_namespace.py

```diff
@@ -0,0 +1,263 @@
+                target_bucket_name = mcg_obj.create_new_namespace_resource(
+                    rand_ns_resource, rand_connection, self.region, cld_mgr, cloud_uls_factory)
+                self.check_ns_resource_validity(mcg_obj, rand_ns_resource,
+                                                target_bucket_name, self.aws_endpoint)
+
+                created_ns_resources.append(rand_ns_resource)
+                created_ns_connections.append(rand_connection)
+            return target_bucket_name, rand_ns_resource
+
+        def ns_resources_and_connections_cleanup():
+            for ns_resource in created_ns_resources:
+                self.delete_ns_resource(ns_resource, mcg_obj)
+
+            for ns_connection in created_ns_connections:
+                self.delete_ns_connection(ns_connection, mcg_obj)
+
+        request.addfinalizer(ns_resources_and_connections_cleanup)
+
+        return _create_ns_resources
+
+    @pytest.fixture()
+    def ns_bucket_factory(self, request, mcg_obj, cld_mgr):
+        """
+        Create a namespace bucket factory. Calling this fixture creates a new namespace bucket(s).
+        For a custom amount, provide the 'amount' parameter.
+
+        Args:
+            mcg_obj (MCG): An MCG object
+            cld_mgr (CM): A cloud manager object containing the MCG S3 connection
+            credentials
+
+        """
+        created_ns_buckets = []
+
+        def _create_ns_buckets(amount=1, write_resource=[], read_resources=[[]], *args, **kwargs):
+            for i in range(amount):
+                # Create random namespace bucket name
+                rand_ns_bucket = create_unique_resource_name('aws', self.ns_bucket)
+
+                # Create actual namespace bucket
+                mcg_obj.create_new_namespace_bucket(rand_ns_bucket, write_resource[i], read_resources[i])
+                created_ns_buckets.append(rand_ns_bucket)
+                self.check_ns_bucket_validity(mcg_obj, rand_ns_bucket, write_resource[i], read_resources[i])
+            return rand_ns_bucket
+
+        def ns_buckets_cleanup():
+            for ns_bucket in created_ns_buckets:
+                self.delete_ns_bucket(ns_bucket, mcg_obj)
+
+        request.addfinalizer(ns_buckets_cleanup)
+
+        return _create_ns_buckets
+
+    def check_ns_resource_validity(self, mcg_obj, ns_resource_name, target_bucket_name, endpoint):
+        """
+        Check namespace resource validity
+        """
+        # Retrieve the NooBaa system information
+        system_state = mcg_obj.send_rpc_query('system_api', 'read_system').json().get('reply')
+
+        # Retrieve the correct namespace resource info
+        for ns_resource in system_state.get('namespace_resources'):
+            if ns_resource.get('name') == ns_resource_name:
+                assert ns_resource.get('target_bucket') == target_bucket_name
+                assert ns_resource.get('endpoint') == endpoint
+                return
+        assert False
```

Comment:
no need to return and no need to assert

Reply:
why not? 
it handles a situation where there are no buckets or the ns_bucket wasn't found as it should be found

Reply:
Got you. So for that, you can do:
```python
    match_resource = [
        ns_resource for ns_resource in system_state
        .get('namespace_resources') if ns_resource.get('name') == ns_resource_name
    ]
    assert match_resource, f"The namespace resource with name {ns_resource_name} was not found"
    assert ns_resource.get('target_bucket') == target_bucket_name
    assert ns_resource.get('endpoint') == endpoint

```

Reply:
just please add assert messages to all the assert statements

Reply:
thanks! 
changed as you suggested and added assert messages :)